src/java/org/apache/cassandra/io/util/SequentialWriter.java

```diff
@@ -386,6 +386,9 @@ public final Throwable commit(Throwable accumulate)
 
     public final Throwable abort(Throwable accumulate)
     {
+        // Do not use this to interrupt a write operation running in another thread.
+        // This is thread-unsafe, releasing and cleaning the buffer while it is being written can have disastrous
+        // consequences (e.g. SIGSEGV).
```

Comment:
Is this comment related to the function or to the return statement? Will it be more appropriate to put the comment in the front of the function?

Reply:
Fixed